EcoSoapBank/Helpers/UtilityFunctions.swift

```diff
@@ -120,3 +120,29 @@ extension DateFormatter {
         $0.timeStyle = .none
     }
 }
+
+/// Alternatively, `?<-`?
+/// (`let x = self.point?.x ?<- CGPoint.zero` vs.
+/// `let x = self.point?.x ??= CGPoint.zero`)
+///
+/// See `Optional.orSettingIfNil` to see use in practice.
+infix operator ??=
+
+extension Optional {
+    /// If nil, sets wrapped value to the new value and then returns it. If non-nil, ignores the new value
+    /// and simply returns the wrapped value.
+    ///
+    /// Similar to the nil-coalescing operator (`??`), but additionally sets the wrapped value if non-nil.
+    mutating func orSettingIfNil(_ newValueIfNil: Wrapped) -> Wrapped {
+        if self == nil { self = newValueIfNil }
+        return self!
+    }
+
+    /// If nil, sets wrapped value to the new value and then returns it. If non-nil, ignores the new value
+    /// and simply returns the wrapped value.
+    ///
+    /// Similar to the nil-coalescing operator (`??`), but additionally sets the left-hand value if non-nil.
+    static func ??=(lhs: inout Wrapped?, rhs: Wrapped) -> Wrapped {
+        lhs.orSettingIfNil(rhs)
+    }
+}
```

Comment:
Very interesting. Not something I would have thought of, but it is easy enough to understand the new operator. I like the operator you chose vs. the arrow one.